filebeat/harvester/log.go

```diff
@@ -180,9 +181,13 @@ func (h *Harvester) sendEvent(event *input.Event) bool {
 // is started. As soon as the output becomes available again, the finished state is written
 // and processing can continue.
 func (h *Harvester) sendStateUpdate() {
+
 	logp.Debug("harvester", "Update state: %s, offset: %v", h.state.Source, h.state.Offset)
 	event := input.NewEvent(h.state)
-	h.outlet.OnEvent(event)
+	h.updateState(event.State)
+
+	data := event.GetData()
+	h.outlet.OnEvent(&data)
```

Comment:
I think that part will stay as we still need to set the state to finished in the registrar state for the reloading feature.